scripts/hostcfgd

```diff
@@ -1669,6 +1671,41 @@ class DnsCfg:
     def dns_update(self, *args, **kwargs):
         run_cmd(['systemctl', 'restart', 'resolv-config'], True, False)
 
+class FipsCfg(object):
+    """
+    FipsCfg Config Daemon
+    Handles the changes in FIPS table.
+    """
+
+    def __init__(self):
+        self.enabled = '0'
+        self.enforced = '0'
+
+    def load(self, data={}):
+        common_config = data.get('global', {})
+        if not common_config:
+            syslog.syslog(syslog.LOG_INFO, f'FipsCfg: skipped the FIPS config, the FIPS setting is empty.')
+            return
+        mode = common_config.get('mode', '').lower()
+        self.enabled = '0'
+        self.enforced = '0'
+        if mode == 'enforce':
+            self.enabled = '1'
+            self.enforced = '1'
+        elif mode == 'noneenforce':
+            self.enabled = '1'
+        self.update()
+
+    def fips_handler():
+        data = self.config_db.get_table('FIPS')
+        self.load(data)
+
+    def update(self):
+        syslog.syslog(syslog.LOG_DEBUG, f'FipsCfg: update fips option enabled: {self.enabled}, enforced: {self.enforced}')
+        try:
+            run_cmd(['sudo', 'update-fips-config', self.enforced, self.enabled])
```

Comment:
>run_cmd [](http://example.com/codeflow?start=12&length=7)

Better to implement in python.

Reply:
Removed update-fips-config, and implemented in python, thanks.